metric_learn/constraints.py

```diff
@@ -80,7 +84,11 @@ def chunks(self, num_chunks=100, chunk_size=2):
     return chunks
 
   @staticmethod
-  def random_subset(all_labels, num_preserved=np.inf):
+  def random_subset(all_labels, num_preserved=np.inf, random_state=None):
+    """
+    the random state object to be passed must be a numpy random seed
+    """
+    state = random_state
     n = len(all_labels)
     num_ignored = max(0, n - num_preserved)
     idx = np.random.randint(n, size=num_ignored)
```

Comment:
The point of passing in a random state is to generate the random numbers we need from it. So in this case, it would look like:

```python
def random_subset(all_labels, num_preserved=np.inf, random_state=np.random):
   ...
   idx = random_state.randint(n, size=num_ignored)
   ...
```

We can use the `np.random` module as the default value, which means it'll call `np.random.randint` by default.
This wouldn't support passing in a raw numeric seed, but that's ok by me: users can just pass in `np.random.RandomState(1234)` if they want to use seed `1234`.